core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -456,11 +467,37 @@ public InputStream fetchLogSegment(final RemoteLogSegmentMetadata remoteLogSegme
                 .toInputStream();
         } catch (final KeyNotFoundException | KeyNotFoundRuntimeException e) {
             throw new RemoteResourceNotFoundException(e);
+        } catch (final ClosedByInterruptException e) {
+            log.debug("Fetching log segment {} with range {} was interrupted", remoteLogSegmentMetadata, range);
+            return InputStream.nullInputStream();
+        } catch (final RuntimeException | StorageBackendException e) {
+            final InputStream is = maybeReturnNullInputStreamIfInterrupted(e, remoteLogSegmentMetadata, range);
+            if (is != null) {
+                return is;
+            } else {
+                throw new RemoteStorageException(e);
```

Comment:
I guess this is covered by `fetchSegmentInterruptionWhenGettingManifest` and `fetchSegmentInterruptionWhenGettingSegment`. Or do you mean something else?

Reply:
Nope, I'm taking about the case when it would be let's say both RuntimeExceptions in the list of arguments in parameterized tests we have now. Basically checking that a RuntimeException will be rethrown in this case. Existining tests are only covering the cases when null input stream should be returned.

Reply:
Gotcha. Please see the last commit